tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/AMQPBlockingTest.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.tests.integration.amqp;
+
+import org.apache.activemq.artemis.core.server.ActiveMQServer;
+import org.apache.activemq.artemis.core.server.files.FileStoreMonitor;
+import org.apache.activemq.artemis.core.server.impl.ActiveMQServerImpl;
+import org.apache.activemq.artemis.core.settings.impl.AddressFullMessagePolicy;
+import org.apache.activemq.artemis.core.settings.impl.AddressSettings;
+import org.apache.activemq.transport.amqp.client.*;
+import org.junit.jupiter.api.Test;
+
+import java.net.URI;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class AMQPBlockingTest extends GlobalDiskFullTest {
+
+   @Override
+   protected ActiveMQServer createServer(int port) throws Exception {
+      ActiveMQServer server = this.createServer(true, true);
+      server.getConfiguration().getAcceptorConfigurations().clear();
+      server.getConfiguration().getAcceptorConfigurations().add(this.addAcceptorConfiguration(server, port));
+      server.getConfiguration().setName("localhost");
+      server.getConfiguration().setJournalDirectory(server.getConfiguration().getJournalDirectory() + port);
+      server.getConfiguration().setBindingsDirectory(server.getConfiguration().getBindingsDirectory() + port);
+      server.getConfiguration().setPagingDirectory(server.getConfiguration().getPagingDirectory() + port);
+      server.getConfiguration().setMessageExpiryScanPeriod(5000L);
+      AddressSettings addressSettings = new AddressSettings();
+      addressSettings.setAddressFullMessagePolicy(AddressFullMessagePolicy.FAIL);
```

Comment:
The Jira indicates an issue with either policy of either FAIL or DROP...but the code change seems like it could only affect FAIL, because its behind a gate checking for FAIL? Is another test and possibly fix needed?

The test method code itself below for FAIL looks essentially identical to the existing GlobalDiskFullTest, which I presume is covering PAGE, so if the same works for DROP then perhaps they could all share the same overall base test, setting the specific mode covered via the constructor (either via ParameterizedTestExtension usage, or actual trivial subclasses of a base class)

Reply:
I'll add a DROP test and see how it behaves.

Reply:
actually the test pass with DROP. The fix only affects FAIL. Looks like the DROP was handled correctly.